src/main/java/org/sonarlint/intellij/util/FutureUtils.java

```diff
@@ -41,12 +44,34 @@ public static void waitForTask(Project project, Future<?> task, String taskName,
     }
   }
 
-  public static void waitForTasks(Project project, List<Future<?>> updateTasks, String taskName) {
+  public static void waitForTasks(Project project, ProgressIndicator indicator, List<Future<?>> updateTasks, String taskName) {
     for (var f : updateTasks) {
-      waitForTask(project, f, taskName, Duration.ofSeconds(20));
+      waitForTask(project, indicator, f, taskName, Duration.ofSeconds(20));
     }
   }
 
+  private static void waitForFuture(ProgressIndicator indicator, Future<?> future, Duration durationTimeout) throws InterruptedException, ExecutionException, TimeoutException {
+    long counter = 0;
+    while (counter < durationTimeout.toMillis()) {
+      counter += 100;
+      if (indicator.isCanceled()) {
+        future.cancel(true);
+        return;
+      }
+      try {
+        future.get(100, TimeUnit.MILLISECONDS);
```

Comment:
`100` is present twice here, maybe use a constant because they should change together